### PR TITLE
Mod/dev 6438 create agency discrepancies endpoint

### PR DESCRIPTION
**Pull request by brett-varney**
**Description:**
create About the Data endpoint /api/v2/reporting/agencies/{agency_code}/discrepancies/ to access data in reporting_agency_missing_tas table

**Technical details:**
standard api with sort/paginng, just that table

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend <OPTIONAL>
    - [x] Operations na new endpoint
    - [x] Domain Expert na
4. [x] Matview impact assessment completed na
5. [x] Frontend impact assessment completed na
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created na
8. [x] Jira Ticket [DEV-6438](https://federal-spending-transparency.atlassian.net/browse/DEV-6438):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison na new data point

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    )
+    mommy.make(
+        "reporting.ReportingAgencyMissingTas",
+        toptier_code=321,
+        fiscal_year=2020,
+        fiscal_period=2,
+        tas_rendering_label="TAS 2",
+        obligated_amount=12.0,
+    )
+
```
Comment (tony-sappe) [inline usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py:36]: Might be worthwhile to create one more record which is never returned in the tests below.

**Review by tony-sappe — CHANGES_REQUESTED** [review]
Overall the PR looks great. Biggest requested change is to fully leverage Tinyshield

File: usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py (hunk 1 above)
Comment (brett-varney) [reply usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py:36]: That one (obligated_amount=12.0) never is.

Comment (tony-sappe) [reply usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py:36]: Oh, thanks. different agency code

File: usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+def test_basic_success(setup_test_data, client):
+    resp = client.get("/api/v2/reporting/agencies/123/discrepancies/?fiscal_year=2020&fiscal_period=3")
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/tests/integration/test_agency_code_discrepancies.py]: changed

File: usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            ReportingAgencyMissingTas.objects.filter(
+                toptier_code=self.toptier_code, fiscal_year=fiscal_year, fiscal_period=fiscal_period
+            )
+            .exclude(obligated_amount=0)
+            .values("tas_rendering_label", "obligated_amount")
+        )
+        results = [
+            {"tas": result["tas_rendering_label"], "amount": result["obligated_amount"]} for result in result_list
+        ]
+        return sorted(results, key=lambda x: x[self.pagination.sort_key], reverse=self.pagination.sort_order == "desc")
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py]: makes sense, done

File: usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        result_list = (
+            ReportingAgencyMissingTas.objects.filter(
+                toptier_code=self.toptier_code, fiscal_year=fiscal_year, fiscal_period=fiscal_period
+            )
+            .exclude(obligated_amount=0)
+            .values("tas_rendering_label", "obligated_amount")
+        )
+        results = [
+            {"tas": result["tas_rendering_label"], "amount": result["obligated_amount"]} for result in result_list
+        ]
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py]: true, done

File: usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                "name": "fiscal_period",
+                "type": "integer",
+                "min": 2,
+                "max": 12,
+                "optional": False,
+                "default": None,
+                "allow_nulls": False,
+            },
+        ]
+        TinyShield(model).block(request.query_params)
```
[the remark this answers is not in the document]
Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py]: fine, done

File: usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    def get_agency_discrepancies(self, fiscal_year, fiscal_period):
+        result_list = (
+            ReportingAgencyMissingTas.objects.filter(
+                toptier_code=self.toptier_code, fiscal_year=fiscal_year, fiscal_period=fiscal_period
+            )
+            .exclude(obligated_amount=0)
+            .values("tas_rendering_label", "obligated_amount")
+            .annotate(tas=F("tas_rendering_label"), amount=F("obligated_amount"))
+        )
+        results = [{"tas": result["tas"], "amount": result["amount"]} for result in result_list]
```
Comment (tony-sappe) [inline usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py:62]: Is this list comprehension necessary?

Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py:62]: Maybe, do you have an alternative that works?

Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py:62]: Oh, I see, I guess not...

Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py:62]: Well, annotation isn't working without including the other columns, which we don't want in the returned results.

Comment (brett-varney) [reply usaspending_api/reporting/v2/views/agencies/agency_code/discrepancies.py]: Nope, got it finally

**Review by tony-sappe — APPROVED** [review]

**Pull request merged**